extension/src/experiments/data/index.ts

```diff
@@ -86,6 +93,58 @@ export class ExperimentsData extends BaseData<ExpShowOutput> {
     this.collectedFiles = collectFiles(data, this.collectedFiles)
   }
 
+  private async collectRevisionGitDetails(
```

Comment:
I created those to reduce complexity, but they're more like helper functions than belonging to a class. Had to keep them in the class because of `this.experiments.getNbOfCommitsToShow(branch)` and `this.internalCommands.executeCommand()`, but arguments shouldn't be moved to private properties as they renew with each  `update` call.